src/full/Agda/Syntax/Translation/ConcreteToAbstract.hs

```diff
@@ -1608,7 +1608,23 @@ instance ToAbstract NiceDeclaration A.Declaration where
           t'  <- toAbstract $ C.Generalized t
           f  <- getConcreteFixity x
           x' <- freshAbstractQName f x
-          bindName' p DataName (GeneralizedVarsMetadata $ generalizeTelVars ls') x x'
+          mErr <- bindName'' p DataName (GeneralizedVarsMetadata $ generalizeTelVars ls') x x'
+          warnings <- useTC stTCWarnings
+          let hasLoneSigWarning = List.any pred warnings
```

Comment:
Intuitively, this way of searching through the warnings does not seem the ideal way how information should flow from the nicifier to the scope checker.  I'd rather see the information attached to the `NiceDataSig` constructor.

To keep Agda maintainable in the long run, we have to keep/make the information flow conspicuous.